logtail: TestConcurrentSameFile is flaky on windows

```
=== RUN   TestConcurrentSameFile
    testing.go:1369: TempDir RemoveAll cleanup: unlinkat C:\Users\RUNNER~1\AppData\Local\Temp\TestConcurrentSameFile100010192\001\testlog.log1.txt: The process cannot access the file because it is being used by another process.
--- FAIL: TestConcurrentSameFile (1.78s)
FAIL
FAIL	tailscale.com/logtail/filch	2.059
```
On Windows, you can't delete open files, which makes this fail.

--- a/logtail/filch/filch_test.go
+++ b/logtail/filch/filch_test.go
@@ -388,7 +388,9 @@ func testMaxFileSize(t *testing.T, replaceStderr bool) {
 func TestConcurrentSameFile(t *testing.T) {
 	filePrefix := filepath.Join(t.TempDir(), "testlog")
 	f1 := must.Get(New(filePrefix, Options{MaxFileSize: 1000}))
+	defer f1.Close()
 	f2 := must.Get(New(filePrefix, Options{MaxFileSize: 1000}))
+	defer f2.Close()
 	var group sync.WaitGroup
 	for _, f := range []*Filch{f1, f2} {
 		group.Go(func() {

